packages/cli/rules/bucket.cursor.md

```diff
@@ -0,0 +1,365 @@
+
+Code Style and Structure
+
+- Write clean, type-safe code when implementing Bucket feature flags
+- Use declarative patterns to check feature state and apply configurations
+- Follow functional programming patterns for feature flag implementation
+- Create reusable hooks and utilities for consistent feature management
+- Document feature flag implementation with clear comments
+- Handle loading and error states properly when checking feature flags
+
+Feature Flag Architecture
+
+- Implement feature flags at the appropriate level of abstraction
+- Use granular permissions with targeted feature rollouts
+- Separate UI components from feature flag logic when possible
+- Create abstraction layers to isolate feature flag implementation details
+- Design feature flags with testability in mind
+- Structure feature flags hierarchically for complex applications
+
+Bucket SDK Usage
+
+- Configure BucketProvider or BucketClient properly at application entry points
+- Use strong typing with TypeScript for feature definitions
+- Properly handle feature loading states to prevent UI flashing
+- Implement proper error fallbacks when feature flag services are unavailable
+- Use environment-specific configuration for development, staging, and production
+- Leverage Bucket CLI for generating type-safe feature definitions
+
+Feature Targeting and Segmentation
+
+- Implement proper user and company context data for targeting
+- Design effective targeting rules based on user attributes
+- Create and manage reusable segments for consistent targeting
+- Use hierarchical targeting rules for complex user populations
+- Monitor and analyze targeting effectiveness
+- Follow best practices for exclusion and inclusion patterns
+
+Remote Configuration
+
+- Define type-safe configuration payloads for feature flags
+- Access and use configuration values correctly in components
+- Design fallback values for configuration when services are unavailable
+- Implement progressive enhancement patterns with remote configuration
+- Test with various configuration scenarios before deployment
+- Use configuration for both functional and visual feature variations
+
+Feature Lifecycle Management
+
+- Follow proper staging processes from development to general availability
+- Document feature flag purpose and intended lifecycle
+- Plan for feature flag retirement and cleanup
+- Implement proper versioning for feature flag definitions
+- Coordinate feature releases across frontend and backend systems
+- Set up proper monitoring for feature adoption and health
+
+Analytics and Feedback
+
+- Track feature usage properly with Bucket analytics
+- Implement feedback collection at appropriate interaction points
+- Design effective user feedback mechanisms for features
+- Analyze feature performance through metrics and KPIs
+- Use STARS framework for effective feature analysis
+- Connect feature metrics with business outcomes
+
+Security and Compliance
```

Comment:
This section can be summarize into one or two statements and merged with `Bucket SDK Usage`